kitura-init.js

```diff
@@ -1,32 +1,148 @@
 #!/usr/bin/env node
 
 const chalk = require('chalk');
-const spawn = require('child_process').spawn;
+const spawnSync = require('child_process').spawnSync;
+const os = require('os');
+const fs = require('fs');
+const path = require('path');
 
-let args = process.argv.slice(2);
+const initURL = 'https://github.com/IBM-Swift/generator-swiftserver-projects'
+const initBranch = 'init'
 
-let options = ['-p', 'yo@1', '-p', 'generator-swiftserver', '--', 'yo', 'swiftserver'];
+let args = process.argv.slice(2);
 
 if (args.length > 0) {
     if (args.indexOf('--help') > -1) {
         printHelp();
         process.exit(0);
     }
+}
+
+let currentDirPath = path.resolve("./");
+let currentDir = path.basename(currentDirPath);
+// Replace spaces with hyphens so Xcode project will build.
+let projName = currentDir.replace(/ /g,"-");
 
-    // Add on the arguments
-    options.push(args);
+if (projName.charAt(0) === '.') {
+    console.error(chalk.red('Application name cannot start with .: %s'));
+   }
+if (projName.toLowerCase() === 'node_modules') {
+    console.error(chalk.red('Application name cannot be {node_modules}'));
+   }
+if (projName.toLowerCase() === 'favicon.ico') {
+    console.error(chalk.red('Application name cannot be {favicon.ico}'));
 }
 
-options.push('--init');
+// Make sure directory is empty.
+checkCurrentDirIsEmpty();
+
+// Clone repo contents into current directory.
+cloneProject(initURL, initBranch);
+
+// Rename project to match current directory
+renameProject();
+
+// If '--skip-build' not specified, then build project.
+if (!(args.includes('--skip-build'))) {
+    buildProject();
+}
+
+function checkCurrentDirIsEmpty() {
+  try {
+      var data = fs.readdirSync('.');
+      if (data.length !== 0) {
+            console.error(chalk.red('Current directory is not empty.'));
+            console.error(chalk.red('Please repeat the command in an empty directory.'));
+            process.exit();
+      }
+  } catch (err) {
+      process.exit();
+  }
+}
+
+function cloneProject(url, branch) {
+    console.log('Creating project...');
+    let clone = spawnSync('git', ['clone', '-b', branch, url, '.']);
+
+    if (clone.status !== 0) {
+        console.error(chalk.red('Error: ') + 'failed to run git clone.');
+        console.error('Please check that you have git installed.');
```

Comment:
Would we also hit this path if the user's network connection is bad (e.g. conference wifi)? Should we also mention checking their network connection?

Reply:
Yes, good idea!